### ISSUE TITLE
Add support for Meta Spatial Anchors OpenXR API

### ISSUE THREAD

**Issue by m4gr3d** [GodotVR/godot_openxr#166]
https://developer.oculus.com/experimental/spatial-anchors-overview/

**Comment by m4gr3d** [comment]
@BastiaanOlij Looks like the api is still experimental so it won't make the `1.2.0` release.

[Experimental features](https://developer.oculus.com/experimental/experimental-overview/) are not allowed in the store and require `adb` commands to enable, so we should wait until it's stable before including.

**Comment by m4gr3d** [comment]
The feature is being developed in the https://github.com/m4gr3d/godot_openxr/tree/dev_spatial_anchor branch.

**Comment by BastiaanOlij** [comment]
@m4gr3d if you want, but it's up to you, I think there is nothing wrong to create a draft PR with the experimental functionality and make it clear in the PR description that this is experimental functionality that can't be used yet in production and that the PR will remain draft until such time that the functionality is made official. This way people can start testing it and start providing feedback.

That is assuming off course that now that the spatial logic is publicaly showcases on the Oculus developer website it no longer falls under any NDA or embargo?

**Comment by m4gr3d** [comment]
> That is assuming off course that now that the spatial logic is publicaly showcases on the Oculus developer website it no longer falls under any NDA or embargo?

Yes, it's publicly available so developers can start prototyping and giving feedback. You just can't ship an app with an experimental api to the Meta store.

**Comment by BastiaanOlij** [comment]
> Yes, it's publicly available so developers can start prototyping and giving feedback. You just can't ship an app with an experimental api to the Meta store.

Then I would recommend creating a draft PR for it so people can provide feedback :) I know atleast two people on the discord who've been hoping this stuff would be implemented :)

**Comment by koenvaneijk** [comment]
> > Yes, it's publicly available so developers can start prototyping and giving feedback. You just can't ship an app with an experimental api to the Meta store.
> 
> Then I would recommend creating a draft PR for it so people can provide feedback :) I know atleast two people on the discord who've been hoping this stuff would be implemented :)

Same, C++ not my language (yet), but happy to support any way I can. Would like to use the SAs in my next game and would hate to have to use Unity for the prototype, much rather would like to use Godot.

Any high-level guidelines would be appreciated or like a to-do with the several components listed, as I have no idea where to begin.

**Comment by m4gr3d** [comment]
Spatial Anchor is [now publicly available](https://developer.oculus.com/documentation/native/android/openxr-lsa-overview/), so it'll be incorporate within an upcoming release.

**Comment by maunvz** [comment]
Is this being actively developed for Godot 4? I started hacking together a GDExtension to support XR_FB_scene, XR_FB_spatial_entity, and related extensions for a personal project, was planning to eventually open source / publish if no one else is already on it

**Comment by m4gr3d** [comment]
> Is this being actively developed for Godot 4? I started hacking together a GDExtension to support XR_FB_scene, XR_FB_spatial_entity, and related extensions for a personal project, was planning to eventually open source / publish if no one else is already on it

@maunvz This repo is in maintenance mode as it's Godot 3 only and we've shifted most of our focus to Godot 4.
For Godot 4, vendor extensions can be added to the https://github.com/GodotVR/godot_openxr_vendors repo.

The repo includes a GDExtension that adds support for [`XR_FB_SCENE_CAPTURE`](https://github.com/GodotVR/godot_openxr_vendors/blob/master/godotopenxrmeta/src/main/cpp/openxr_fb_scene_capture_extension_wrapper.cpp) so you could add to it!
